### PR TITLE
Remove block on deprecated_warnings. 

### PR DESCRIPTION
**Pull request by kjmcqueen**
Update VM signalHandler to modern standards.

https://rchain.atlassian.net/browse/ROS-302

### CODE REVIEW

File: rosette/src/Vm.cc [hunk 1]
```diff
@@ -503,7 +497,16 @@ void VirtualMachine::handleSleep() {
...
 
 void VirtualMachine::handleSignal() {
-    int oldmask = sigblock(IO_MASK);
+    sigset_t oldmask;
+    sigset_t blockmask;
+
+    sigemptyset(&blockmask);
+    sigaddset(&blockmask, SIGIO);
+    if (sigprocmask(SIG_BLOCK, &blockmask, &oldmask) < 0) {
+        printf("aborting...\n");
```
Comment (nashef) [inline rosette/src/Vm.cc]: `fprintf(stderr, "...")` perhaps? Also, "aborting..." is a pretty lame error message unless something was being printed before that? Can you upgrade the error?

File: rosette/src/Vm.cc [hunk 2]
```diff
@@ -512,7 +515,7 @@ void VirtualMachine::handleSignal() {
     if (sigvec & sigmask(SIGINT)) {
         printf("aborting...\n");
         reset();                   /* clears sigvec */
-        (void)sigsetmask(oldmask); /* enable interrupts */
+        (void)sigprocmask(SIG_SETMASK, &oldmask, NULL); /* enable interrupts */
         return;
     }
```
Comment (nashef) [inline rosette/src/Vm.cc]: Please add a TODO here and then file a bug to get rid of the `goto`. Holy hell.

**Comment by nashef** [comment]
Your change looks like a Pareto improvement on the old code, but if I'm reading this correctly it needs a lot more work. I think it would be very hard to make this call reentrant enough to be safe to call `select` from a signal handler. Can you confirm whether or not this is the actual handler or whether it's just an event handler that eventually fires when a signal is sent elsewhere? It matters, because on modern Unix it won't work reliably if it's the former.

**Comment by nashef** [comment]
I only see the three uses in `handleSignal`, myself.

**Comment by kirkwood** [comment]
Then I've done something weird:
```
[ 73%] Building CXX object src/CMakeFiles/rosette.dir/Vm.cc.o
cd /tmp/rchain/rosette/build.out/src && /usr/bin/c++   -I/tmp/rchain/rosette/h -I/tmp/rchain/rosette/h/sys  -std=c++11 -DGCC27X -ggdb -m32 -Wall -Werror -Wno-write-strings -Wno-invalid-offsetof -Wno-delete-incomplete   -o CMakeFiles/rosette.dir/Vm.cc.o -c /tmp/rchain/rosette/src/Vm.cc
/tmp/rchain/rosette/src/Vm.cc: In member function ‘void VirtualMachine::handleSignal()’:
/tmp/rchain/rosette/src/Vm.cc:522:18: error: ‘IO_MASK’ was not declared in this scope
     if (sigvec & IO_MASK) {
                  ^~~~~~~
/tmp/rchain/rosette/src/Vm.cc:522:18: note: suggested alternative: ‘IOV_MAX’
     if (sigvec & IO_MASK) {
                  ^~~~~~~
                  IOV_MAX
```

**Comment by kjmcqueen** [comment]
I fixed the IO_MASK usages.  I don't know how it worked as I had removed the definition earlier.

File: rosette/src/Vm.cc (hunk 2 above)
Comment (kjmcqueen) [reply rosette/src/Vm.cc]: It doesn't appear that this is the actual signal handler.  The actual handler is in Interrupt.cc RosetteSignalHandler()  which calls vm->acceptSignal with the signal raised which in turn stores it in the sigvec variable.

handleSignal() is called from what I think is the main loop in VirtualMachine::getNextStrand() .

**Comment by kjmcqueen** [comment]
It doesn't appear that this is the actual signal handler. The actual handler is in Interrupt.cc RosetteSignalHandler() which calls vm->acceptSignal with the signal raised which in turn stores it in the sigvec variable.

handleSignal() is called from what I think is the main loop in VirtualMachine::getNextStrand() .

**Comment by nashef** [comment]
`136     void acceptSignal(int sig) { sigvec |= sigmask(sig); }`

Reeeeeeeeeeeeeeeeeeeeeeee.

Ok, I'm over it. Your change LGTM if it passes tests.

**Pull request merged**